pyuvdata/uvdata/initializers.py

```diff
@@ -0,0 +1,579 @@
+
+    if channel_width is None:
+        channel_width = freq_array[1:] - freq_array[:-1]
+        channel_width = np.concatenate([channel_width, channel_width[-1:]])
+    elif np.isscalar(channel_width):
+        channel_width = np.full_like(freq_array, channel_width)
+
+    if not isinstance(channel_width, np.ndarray):
+        raise ValueError("channel_width must be a numpy array or a scalar.")
+
+    if channel_width.shape != freq_array.shape:
+        raise ValueError("channel_width must be the same shape as freq_array.")
+
+    return freq_array, channel_width
+
+
+def get_baseline_params(
+    antenna_positions: np.ndarray,
+    antpairs: Sequence[tuple[int, int]] | None = None,
+    unique_antpairs: Sequence[tuple[int, int]] | None = None,
+) -> tuple[np.ndarray, np.ndarray]:
+    """Configure baseline parameters for new UVData object."""
+    if unique_antpairs is None and antpairs is None:
+        raise ValueError("Either antpairs or unique_antpairs must be provided.")
+
+    if unique_antpairs is None:
+        ant_1_array, ant_2_array = zip(*antpairs)
+        baseline_array = utils.antnums_to_baseline(
+            ant_1_array, ant_2_array, len(antenna_positions)
+        )
+        unique_baselines = None
+    else:
+        ant_1_array, ant_2_array = zip(*unique_antpairs)
+        unique_baselines = utils.antnums_to_baseline(
+            ant_1_array, ant_2_array, len(antenna_positions)
+        )
+        baseline_array = None
+
+    return baseline_array, unique_baselines
+
+
+def configure_blt_rectangularity(
+    time_array: np.ndarray | None = None,
+    unique_times: np.ndarray | None = None,
+    baseline_array: np.ndarray | None = None,
+    unique_baselines: np.ndarray | None = None,
+    blts_are_rectangular: bool | None = None,
+    time_axis_faster_than_bls: bool | None = None,
+):
+    """Configure blt rectangularity parameters for new UVData object."""
+    if unique_baselines is None and baseline_array is None:
+        raise ValueError("Either baseline_array or unique_baselines must be provided.")
+
+    if unique_times is None and time_array is None:
+        raise ValueError("Either time_array or unique_times must be provided.")
+
+    if unique_baselines is not None and baseline_array is not None:
+        raise ValueError(
+            "Only one of baseline_array or unique_baselines can be provided."
+        )
+
+    if unique_times is not None and time_array is not None:
+        raise ValueError("Only one of time_array or unique_times can be provided.")
+
+    if unique_times is not None and baseline_array is not None:
+        raise ValueError(
+            "If unique_times are provided, unique_baselines must be provided "
+            "(not baseline_array)."
+        )
+
+    if unique_baselines is not None and time_array is not None:
+        raise ValueError(
+            "If unique_baselines are provided, unique_times must be provided as well "
+            "(not time_array)."
+        )
```

Comment:
Yeah, good point. Actually, I refactored all this bit of the code to avoid this, and I think to make it a simpler interface. Now, you don't pass either `unique_times` or `time_array`, you just pass `times` and specify `do_blt_outer`. There's some logic to figure out what `do_blt_outer` is by default based on the passed times/antpairs, but you can shortcircuit that by just providing it.